tests/009-curl_exec.phpt

```diff
@@ -0,0 +1,31 @@
+--TEST--
+Calls to curl_exec are logged
+--SKIPIF--
+<?php if (!extension_loaded("scoutapm")) die("Skipped: scoutapm extension required."); ?>
+<?php if (!extension_loaded("curl")) die("Skipped: curl extension required."); ?>
+--FILE--
+<?php
+$ch = curl_init();
+curl_setopt($ch, CURLOPT_URL, "file://" . __FILE__);
+curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+curl_exec($ch);
+
+$call = scoutapm_get_calls()[0];
+
+var_dump($call['function']);
+var_dump($call['entered']);
+var_dump($call['exited']);
+var_dump($call['time_taken']);
+var_dump($call['exited'] > $call['entered']);
+var_dump($call['argv']);
+?>
+--EXPECTF--
+string(9) "curl_exec"
+float(%f)
+float(%f)
+float(%f)
+bool(true)
+array(1) {
+  [0]=>
+  resource(4) of type (curl)
```

Comment:
it's worth pointing out that the args here are largely pointless; the URL for a `curl` request is actually set in the `curl_setopt` call, but it's `curl_exec` that will take time (so should be logged). It's not immediately obvious how to track this back. We'll have the same issue with PDO and fread/fwrite too. I may open a separate issue to improve on this though.

Reply:
Unfortunately, I don't think it's that simple. `curl` in PHP is just a proxy really to `libcurl`, which doesn't appear to provide a way of getting arguments out again. `$ch` is not an "object", but a "resource" which are mostly evil magic in PHP. I think we can handle it; but we'd have to do some special logic to handle these cases. At the moment, I've simplified to such an extent that all the handlers we override are actually just one simple handler (reduces the size of the binary), but we may need to add some extra checks in there. The important point is that we're measuring the functions that perform the actual I/O, the enhancement is to ensure we can see what the I/O actually is. I'll open a separate issue to handle these I think.

Reply:
Opened #28 to handle this.